cleanrl/ppo_dna_atari_envpool.py

```diff
@@ -0,0 +1,468 @@
+# DNA: Proximal Policy Optimization with a Dual Network Architecture
+# https://arxiv.org/abs/2206.10027
+
+import argparse
+import os
+import random
+import time
+from collections import deque
+from distutils.util import strtobool
+
+import envpool
+import gym
+import numpy as np
+import torch
+import torch.nn as nn
+import torch.optim as optim
+from torch.distributions import Categorical, kl_divergence
+from torch.utils.tensorboard import SummaryWriter
+
+
+def parse_args():
+    # fmt: off
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--exp-name", type=str, default=os.path.basename(__file__).rstrip(".py"),
+        help="the name of this experiment")
+    parser.add_argument("--seed", type=int, default=np.random.randint(2 ** 31),
+        help="seed of the experiment")
+    parser.add_argument("--torch-deterministic", type=lambda x: bool(strtobool(x)), default=True, nargs="?", const=True,
+        help="if toggled, `torch.backends.cudnn.deterministic=False`")
+    parser.add_argument("--cuda", type=lambda x: bool(strtobool(x)), default=True, nargs="?", const=True,
+        help="if toggled, cuda will be enabled by default")
+    parser.add_argument("--track", type=lambda x: bool(strtobool(x)), default=False, nargs="?", const=True,
+        help="if toggled, this experiment will be tracked with Weights and Biases")
+    parser.add_argument("--wandb-project-name", type=str, default="cleanRL",
+        help="the wandb's project name")
+    parser.add_argument("--wandb-entity", type=str, default=None,
+        help="the entity (team) of wandb's project")
+
+    # Algorithm specific arguments
+    parser.add_argument("--env-id", type=str, default="Pong-v5",
+        help="the id of the environment")
+    parser.add_argument("--total-timesteps", type=int, default=10_000_000,
+        help="total timesteps of the experiments")
+    parser.add_argument("--num-envs", type=int, default=128,
+        help="the number of parallel game environments")
+    parser.add_argument("--envpool-num-threads", type=int, default=2,
+        help="the number of envpool worker threads")
+    parser.add_argument("--num-steps", type=int, default=128,
+        help="rollout horizon: the number of steps to run in each environment per policy rollout")
+    parser.add_argument("--anneal-lr", type=lambda x: bool(strtobool(x)), default=False, nargs="?", const=True,
+        help="Toggle learning rate annealing for policy and value networks")
+    parser.add_argument("--gamma", type=float, default=0.999,
+        help="the discount factor gamma")
+
+    parser.add_argument("--policy-learning-rate", type=float, default=2.5e-4,
+        help="the learning rate of the policy optimizer")
+    parser.add_argument("--policy-batch-size", type=int, default=2048,
```

Comment:
put a note here

Reply:
I don't follow, you mean a comment explaining why there is an empty line? or something else?

Reply:
Sorry, that sentence was mainly for me. My original thought was that the other PPO related scripts have used `--num-minibatches`, so in this case we might be able to change it to `--num-policy-minibatches`. I will let you decide on this though. Feel free to pick one that feels like the cleanest design to you.

Reply:
Ah true; I changed the config to use minibatch size, to be able to easily match/verify the values against the the paper